### PR TITLE
User first last name UI

### PR DESCRIPTION
**Pull request by ousmaneo**
These changes, add first and last name fields to the user create and edit form and remove the fullName field.
Fix: Graylog2/graylog-plugin-cloud#740

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

/jenkins-pr-deps Graylog2/graylog-plugin-enterprise#2054

### CODE REVIEW

File: graylog2-web-interface/src/components/users/UserEdit/ProfileUpdateInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+
+import { Alert } from 'components/graylog';
+import { Icon } from 'components/common';
+
+const PermissionsUpdateInfo = () => (
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/users/UserEdit/ProfileUpdateInfo.tsx:22]: We should rename the component to `ProfileUpdateInfo`.

File: graylog2-web-interface/src/components/users/UserEdit/ProfileUpdateInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  <Alert bsStyle="info">
+    <Icon name="info-circle" />{' '}<b> First and Last Name</b><br />
+    With Graylog 4.1, we&apos;ve added distinct First Name and Last Name fields. These must be provided before the user’s Profile can be saved.
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/users/UserEdit/ProfileUpdateInfo.tsx:25]:
```suggestion
    With Graylog 4.1, we&apos;ve added distinct first and last name fields. These must be provided before the user’s profile can be saved.
```

**Review by edmundoa — CHANGES_REQUESTED** [review]
So far I haven't tested it but saw a couple smaller things while looking at the code.

File: graylog2-web-interface/src/components/users/UserEdit/ProfileUpdateInfo.tsx (hunk 1 above)
Comment (ousmaneo) [reply graylog2-web-interface/src/components/users/UserEdit/ProfileUpdateInfo.tsx]: My mistake. Fixed it

**Review by danotorrey — COMMENTED** [review]
Should this message only show when the user had not previously specified the first and last name? Similar to how the Full Name field is conditionally displayed. This might be good, since it's not relevant for users who were newly created with a first and last name.

image

**Comment by ousmaneo** [comment]
Thanks @dan, this was astually a small bug. Fixed it

**Review by danotorrey — APPROVED** [review]
LGTM! I'll let @edmundoa do the final verification and merge it.

**Comment by thll** [comment]
@ousmaneo: while playing around with this, I've noticed the following messages when creating a user with a first name and a last name provided:

image

**Comment by ousmaneo** [comment]
Thank you @thll ,  fixed it!

**Review by edmundoa — COMMENTED** [review]
The changes look good to me but I'm not sure why the build is failing. I added the dependency to Graylog2/graylog-plugin-enterprise#2054 in the PR description, but maybe we need to update the code with the changes in master.

**Review by edmundoa — APPROVED** [review]
Since the base branch has the same build errors as this one right now, I guess the error is not coming from this PR but some outdated code base. I'll approve now and we can still fix anything in the other branch if needed before merging to master.

**Comment by ousmaneo** [comment]
Yes, it's mostly outdated code. Thank you @edmundoa 👍

**Pull request merged**